hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotLocalDataYaml.java

```diff
@@ -137,10 +137,12 @@ public Object construct(Node node) {
         // Set other fields from parsed YAML
         snapshotLocalData.setSstFiltered((Boolean) nodes.getOrDefault(OzoneConsts.OM_SLD_IS_SST_FILTERED, false));
 
+        // Handle potential Integer/Long type mismatch from YAML parsing
+        Object lastCompactionTimeObj = nodes.getOrDefault(OzoneConsts.OM_SLD_LAST_COMPACTION_TIME, -1L);
```

Comment:
Suggestion to harden the YARML parsing.

Object obj = nodes.getOrDefault(OzoneConsts.OM_SLD_LAST_COMPACTION_TIME, -1L);
long lastCompactionTime;
if (obj instanceof Number) {
    lastCompactionTime = ((Number) obj).longValue();
} else {
    // fallback or error
    lastCompactionTime = -1L;
}